src/js/common/schemaform/save-in-progress/RoutedSavableApp.jsx

```diff
@@ -0,0 +1,237 @@
+      // Chrome requires this to be set
+      e.returnValue = message;     // eslint-disable-line no-param-reassign
+    }
+    return message;
+  }
+
+  redirectOrLoad(props) {
+    // Stop a user that's been redirected to be redirected again after logging in
+    this.shouldRedirectOrLoad = false;
+
+
+    const firstPagePath = props.routes[props.routes.length - 1].pageList[0].path;
+    // If we're logged in and have a saved / pre-filled form, load that
+    if (props.isLoggedIn) {
+      const currentForm = props.formConfig.formId;
+      const isSaved = props.savedForms.some((savedForm) => savedForm.form === currentForm);
+      const hasPrefillData = props.prefillsAvailable.includes(currentForm);
+      if (isSaved || hasPrefillData) {
+        props.fetchInProgressForm(currentForm, props.formConfig.migrations, !isSaved && hasPrefillData);
+      } else {
+        // No forms to load; go to the beginning
+        // If the first page is not the intro and uses `depends`, this will probably break
+        props.router.replace(firstPagePath);
+      }
+    } else {
+      // Can't load a form; go to the beginning
+      // If the first page is not the intro and uses `depends`, this will probably break
+      props.router.replace(firstPagePath);
+    }
+  }
+
+  removeOnbeforeunload = () => {
+    window.removeEventListener('beforeunload', this.onbeforeunload);
+  }
+
+  render() {
+    const { currentLocation, formConfig, children, formData, loadedStatus } = this.props;
+    const trimmedPathname = currentLocation.pathname.replace(/\/$/, '');
+    const isIntroductionPage = trimmedPathname.endsWith('introduction');
+    let content;
+    const loadingForm = trimmedPathname.endsWith('resume') || loadedStatus === LOAD_STATUSES.pending;
+    if (!formConfig.disableSave && loadingForm) {
+      content = <LoadingIndicator message="Retrieving your saved form..."/>;
+    } else if (!formConfig.disableSave && this.props.savedStatus === SAVE_STATUSES.pending) {
+      content = <LoadingIndicator message="Saving your form..."/>;
+    } else if (!formConfig.disableSave && this.shouldRedirectOrLoad) {
+      content = <LoadingIndicator message="Retrieving your profile information..."/>;
+    } else if (!isInProgress(trimmedPathname)) {
+      content = children;
+    } else {
+      content = (
+        <div>
+          <FormNav formData={formData} formConfig={formConfig} currentPath={trimmedPathname}/>
+          <div className="progress-box progress-box-schemaform">
+            {children}
+          </div>
+        </div>
+      );
+    }
+
+    return (
+      <div>
+        <Element name="topScrollElement"/>
+        {
+          formConfig.title &&
+          // If we’re on the introduction page, show the title if we’re actually on the loading screen
+          (!isIntroductionPage || (!formConfig.disableSave && this.props.loadedStatus !== LOAD_STATUSES.notAttempted)) &&
```

Comment:
@jbalboni I updated this PR with the change you just merged

Reply:
I think disableSave will always be true in here, won't it?

Reply:
Oh yeah you're totally right, can remove this then